python/ray/air/util/tensor_extensions/arrow.py

```diff
@@ -470,12 +470,7 @@ def __getitem__(self, key):
         # unfortunately overriding Cython cdef methods with normal Python methods isn't
         # allowed.
         if isinstance(key, slice):
-            sliced = super().__getitem__(key).to_numpy()
-            if sliced.dtype.type is not np.object_:
-                # Force ths slice to match NumPy semantics for unit (single-element)
-                # slices.
-                sliced = sliced[0:1]
-            return sliced
+            return super().__getitem__(key)
```

Comment:
The Arrow extension array contract is generally to another extension array for a slice, and the element type (in our case an ndarray) when getting a single row.